realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -293,6 +297,29 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Waits for change committed by {@link SharedGroup} in other Thread.
+     *
+     * @return {@code true} if successfully detects change, {@code false} no change has detected otherwise.
+     */
+    public boolean waitForChange() {
+        if (isStoppedWaitingByUser) {
+            isStoppedWaitingByUser = false;
+            return false;
+        }
```

Comment:
No, `isStoppedWaitingByUser` won't work as expected. two problems:
Let's say the timing of calling `stopWaitForChange()` is here the line i made comments, then the exact same problem will happen as the one it has before add `isStoppedWaitingByUser`.